src/main/java/com/aws/greengrass/cisclient/ConnectivityInfoProvider.java

```diff
@@ -33,6 +38,12 @@ public class ConnectivityInfoProvider {
 
     private volatile List<String> cachedHostAddresses = Collections.emptyList();
 
+    public static final int CLIENT_RETRY_COUNT = 3;
+    private static final RetryUtils.RetryConfig CLIENT_EXCEPTION_RETRY_CONFIG =
+            RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofSeconds(30))
+                    .maxRetryInterval(Duration.ofSeconds(30)).maxAttempt(CLIENT_RETRY_COUNT)
+                    .retryableExceptions(Arrays.asList(DeviceConfigurationException.class)).build();
```

Comment:
I don't think we should be retrying this, it isn't going to fix itself. Retries should be handled at a higher level.

Reply:
Retrying to cover the edge case where client is requested at the same time configs are being updated, so client returns null, but an immediate retry would be successful.

Reply:
That "immediate" retry will be in 30 seconds as per this though, right? That sounds too much, so does the 3 retries.

Reply:
What would be reasonable? (5 sec and 2 retries?)

Reply:
Retries are handled at another layer. Don't retry anything here. Just fail